### PR TITLE
feat: [IDP-156]: added RBAC configs for IDP Admin

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Describe your changes
Added new Permission IDP_ADMIN and related configs
Relevant docs [here](https://harness.atlassian.net/wiki/spaces/IDP/pages/21282816244/Access+Control+Tech+Spec+WIP)

##Jira Links
[IDP-156](https://harness.atlassian.net/browse/IDP-156)

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[IDP-156]: https://harness.atlassian.net/browse/IDP-156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43667)

### CODE REVIEW

File: platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/idp/IDPResourceImpl.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public List<Boolean> validate(List<String> resourceIds, Scope scope) {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public ImmutableMap<ScopeLevel, EnumSet<ValidatorType>> getSelectorKind() {
+        return ImmutableMap.of(ScopeLevel.ACCOUNT, EnumSet.of(BY_RESOURCE_TYPE, BY_RESOURCE_TYPE_INCLUDING_CHILD_SCOPES));
```
Comment (sandeepa-kv-jois) [inline platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/idp/IDPResourceImpl.java:63]: We need to confirm BY_RESOURCE_IDENTIFIER, BY_ATTRIBUTE_INCLUDING_CHILD_SCOPES and BY_ATTRIBUTE are not needed for our use case

**Comment by sathish-soundarapandian** [comment]
trigger ti1

**Comment by sarthakkasat** [comment]
trigger ti1

**Comment by sarthakkasat** [comment]
trigger ti1

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43667)* status: Some of the checks are too old. 11 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1389 at r4](https://reviewable.io/reviews/harness/harness-core/43667#-NO8LiPBCZS2ErWgd6Ga:-NO8LiPBCZS2ErWgd6Gb:b-b9l18b) ([raw file](https://github.com/harness/harness-core/blob/793fcff60a3482fe120a5a959977151812ccdcb3/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1389)):*
> ```YAML
>     allowedScopeLevels:
>       - account
>   - identifier: idp_admin_manage
> ```

Its not clear from this identifier what it is doing.

idp_policy_manage? maybe
___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/resources/resourceTypes.yml` line 127 at r4](https://reviewable.io/reviews/harness/harness-core/43667#-NO8Ld189iStzpl2-kHE:-NO8Ld189iStzpl2-kHF:b-8dugbd) ([raw file](https://github.com/harness/harness-core/blob/793fcff60a3482fe120a5a959977151812ccdcb3/access-control/service/src/main/resources/io/harness/accesscontrol/resources/resourceTypes.yml#L127)):*
> ```YAML
>   - identifier: STREAMING_DESTINATION
>     permissionKey: streamingDestination
>   - identifier: IDP_ADMIN
> ```

IDP_ADMIN doesn't sound like a resource

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43667)* status: Some of the checks are too old. 11 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/resources/resourceTypes.yml` line 127 at r4](https://reviewable.io/reviews/harness/harness-core/43667#-NO85dos7xyC_cnbEWd0:-NO85dos7xyC_cnbEWd1:b-nt1pb5) ([raw file](https://github.com/harness/harness-core/blob/793fcff60a3482fe120a5a959977151812ccdcb3/access-control/service/src/main/resources/io/harness/accesscontrol/resources/resourceTypes.yml#L127)):*
> ```YAML
>   - identifier: STREAMING_DESTINATION
>     permissionKey: streamingDestination
>   - identifier: IDP_ADMIN
> ```

Is the resource type really IDP_ADMIN? Can you provide more context on same?



_Code quote:_
```YAML
IDP_ADMIN
```

**Comment by sarthakkasat** [comment]
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1389 at r4](https://reviewable.io/reviews/harness/harness-core/43667#-NO8LiPBCZS2ErWgd6Ga:-NO8VsKs9cvsd7lXv4aI:bpyo9e5) ([raw file](https://github.com/harness/harness-core/blob/793fcff60a3482fe120a5a959977151812ccdcb3/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1389)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Its not clear from this identifier what it is doing.

idp_policy_manage? maybe
</blockquote></details>

Initially I had used idp_policy_manage but I had a discussion with Abhinav Rastogi he said that this might confuse with the governance module, this role only allows to manage the IDP Settings UI page and from  IDP Settings UI page we will select the the policies.
 https://github.com/harness/harness-core-ui/pull/13944#discussion_r1098370017

**Comment by sarthakkasat** [comment]
trigger ti1

**Comment by sarthakkasat** [comment]
*[`access-control/service/src/main/resources/io/harness/accesscontrol/resources/resourceTypes.yml` line 127 at r4](https://reviewable.io/reviews/harness/harness-core/43667#-NO85dos7xyC_cnbEWd0:-NO8usIv-u-Zq-Qt2YZb:bdzvknj) ([raw file](https://github.com/harness/harness-core/blob/793fcff60a3482fe120a5a959977151812ccdcb3/access-control/service/src/main/resources/io/harness/accesscontrol/resources/resourceTypes.yml#L127)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Is the resource type really IDP_ADMIN? Can you provide more context on same?
</blockquote></details>

The IDP_ADMIN role will just allow users to select policies(set of permissions) from the settings UI of IDP-Service

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43667)* status: 11 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1389 at r4](https://reviewable.io/reviews/harness/harness-core/43667#-NO8LiPBCZS2ErWgd6Ga:-NODI7Ej83YHr08WhCuL:b-krl6vf) ([raw file](https://github.com/harness/harness-core/blob/dc9c7355f7900d7d4ba486e66489ea89280d4a1c/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1389)):*
<details><summary><i>Previously, sarthakkasat (Sarthak Kasat) wrote…</i></summary><blockquote>

Initially I had used idp_policy_manage but I had a discussion with Abhinav Rastogi he said that this might confuse with the governance module, this role only allows to manage the IDP Settings UI page and from  IDP Settings UI page we will select the the policies.
 https://github.com/harness/harness-core-ui/pull/13944#discussion_r1098370017
</blockquote></details>

I am not convinced about this permission name. Can we check with PM. I think there should not be any confusion because idp is a separate module

**Comment by abhinavrastogi-harness** [comment]
@ankit-harness @sarthakkasat @Jimit09Gandhi `IDP_POLICY` is not the right name for this permission. It is not controlling access to any "policy" resource. This permission is controlling who can change any setting in IDP admin UI, which includes enabling/disabling plugins, changing layout, managing rbac policies etc.

I agree `IDP_ADMIN` is also not ideal since `ADMIN` is not a resource.. any other suggestions would be welcome.

File: access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml [hunk 2]
```diff
@@ -1384,5 +1384,11 @@ permissions:
   - identifier: core_streamingDestination_delete
     name: Delete Streaming Destination
     status: ACTIVE
+    allowedScopeLevels:
+      - account
+  - identifier: idp_idpsettings_manage
```
Comment (reetika04) [inline access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:1389]: What about the view permission?

Comment (reetika04) [reply access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:1389]: Please add this permission to account admin in managed-roles.yml

Comment (sarthakkasat) [reply access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:1389]: The view permission in by default and I have added the permission for Account Admin in [managed-roles.yaml](https://github.com/harness/harness-core/pull/43667/files#diff-d94b5763a2d9102355354b57084906019c7aea0a6840d34f53d367ff6d5b504f)

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43667)* status: Some of the checks are too old. 9 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1389 at r5](https://reviewable.io/reviews/harness/harness-core/43667#-NNkD615o99Ef31lftAf-r5-1389:-NOoAYkCBgv9g1-V94wa:b-1z63hr) ([raw file](https://github.com/harness/harness-core/blob/5296f38d774223c29f7cff62922974e16ca2474a/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1389)):*
<details><summary><i>Previously, sarthakkasat (Sarthak Kasat) wrote…</i></summary><blockquote>

The view permission in by default and I have added the permission for Account Admin in [managed-roles.yml](https://github.com/harness/harness-core/pull/43667/files#diff-d94b5763a2d9102355354b57084906019c7aea0a6840d34f53d367ff6d5b504f)
</blockquote></details>

@sarthakkasat  Do you mean there is no explicit view permission required?
___
*[`platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/idp/IDPResourceImpl.java` line 40 at r5](https://reviewable.io/reviews/harness/harness-core/43667#-NOoBmP6D3s9m8PQGMFJ:-NOoBmP6D3s9m8PQGMFK:b-2n2equ) ([raw file](https://github.com/harness/harness-core/blob/5296f38d774223c29f7cff62922974e16ca2474a/platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/idp/IDPResourceImpl.java#L40)):*
> ```Java
> @Slf4j
> public class IDPResourceImpl implements Resource {
>   public static final String IDP = "IDP_ADMIN";
> ```

Please change this to have same resource name IDP_SETTINGS.

**Comment by sarthakkasat** [comment]
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1389 at r5](https://reviewable.io/reviews/harness/harness-core/43667#-NNkD615o99Ef31lftAf-r5-1389:-NOoGHe88mKuKff_GVEL:bhiw1qt) ([raw file](https://github.com/harness/harness-core/blob/5296f38d774223c29f7cff62922974e16ca2474a/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1389)):*
> @sarthakkasat  Do you mean there is no explicit view permission required?
Yes this permission will only allow to manage the permissions(backstage specific permissions) in settings UI of IDP-Service but that page will be visible to everyone hence there is no explicit view permission

**Comment by sarthakkasat** [comment]
*[`platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/idp/IDPResourceImpl.java` line 40 at r5](https://reviewable.io/reviews/harness/harness-core/43667#-NOoBmP6D3s9m8PQGMFJ:-NOoMYqV-A0jfUN-tdB3:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5296f38d774223c29f7cff62922974e16ca2474a/platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/idp/IDPResourceImpl.java#L40)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Please change this to have same resource name IDP_SETTINGS.
</blockquote></details>

Done.



_Code quote:_
```Java
  public ImmutableMap<ScopeLevel, EnumSet<ValidatorType>> getSelectorKind() {
    return ImmutableMap.of(ScopeLevel.ACCOUNT, EnumSet.of(BY_RESOURCE_TYPE, BY_RESOURCE_TYPE_INCLUDING_CHILD_SCOPES));
```

**Comment by sarthakkasat** [comment]
trigger gitleaks

**Comment by sarthakkasat** [comment]
trigger gitleaks

**Comment by sarthakkasat** [comment]
trigger gitleaks

**Comment by sarthakkasat** [comment]
trigger ti1

**Pull request closed without merging**